src/main/java/org/stellar/sdk/requests/SSEUtils.java

```diff
@@ -0,0 +1,72 @@
+package org.stellar.sdk.requests;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.internal.sse.RealEventSource;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSourceListener;
+import org.stellar.sdk.responses.GsonSingleton;
+
+import javax.annotation.Nullable;
+import java.net.SocketException;
+
+
+public class SSEUtils {
+  public static <T extends org.stellar.sdk.responses.Response> SSEManager<T> stream(final OkHttpClient okHttpClient, final RequestBuilder requestBuilder, final Class<T> clazz, final EventListener<T> listener) {
+    SSEManager<T> sseManager = new SSEManager<T>(okHttpClient, requestBuilder, clazz, listener);
+    sseManager.start();
+    return sseManager;
+  }
+
+  static <T extends org.stellar.sdk.responses.Response> EventSource streamInternal(final OkHttpClient okHttpClient, final RequestBuilder requestBuilder, final Class<T> clazz, final EventListener<T> listener, String url, final CloseListener closeListener ) {
+
+    Request request = new Request.Builder()
+            .url(url)
+            .header("Accept","text/event-stream")
+            .build();
+    RealEventSource eventSource = new RealEventSource(request, new EventSourceListener() {
+
+      @Override
+      public void onClosed(EventSource eventSource) {
+        if(closeListener!=null) {
+          closeListener.closed(eventSource);
+        }
+      }
+
+      @Override
+      public void onOpen(EventSource eventSource, Response response) {
+      }
+
+      @Override
+      public void onFailure(EventSource eventSource, @Nullable Throwable t, @Nullable Response response) {
+        int code = -1;
+        if(response != null) {
+          code = response.code();
+        }
+        if(t!=null) {
+          if(t instanceof SocketException) {
+            // not a failure, server disconnected
+          } else {
+            throw new IllegalStateException("Failed " + code, t);
+          }
+        } else {
+          throw new IllegalStateException("Failed " + code);
+        }
+      }
+
+      @Override
+      public void onEvent(EventSource eventSource, @Nullable String id, @Nullable String type, String data) {
+        if (data.equals("\"hello\"")) {
```

Comment:
Horizon sends 4 types of messages:
* `event:open`
* `event:close`
* `event:err`

`event` field is empty when sending data. So you should ignore all `open` and `close` events.

Reply:
this should be handled by okhttp-sse EventSourceListener. If I got this wrong, we should be seeing parse errors. I think jersey did something similar under the hood. This hello messages is always the first thing on the wire. After that you get proper json.

Reply:
Horizon will also send [`goodbye`](https://github.com/stellar/go/blob/b3cacf3189c40fc2ff68bd45fadd5716b9ccc83f/services/horizon/internal/render/sse/main.go#L89) event before closing a stream so this must be handled.

Reply:
this is handled now